### ISSUE TITLE
Background service not running on first call

### ISSUE THREAD

**Issue by donettech** [Almoullim/background_location#103]
I implemented this package to get background location, However, the service won't start on the first call and performs normally on the second and following calls.

`ElevatedButton(
child:Text("Start),
onPressed()async{
                     await bg.BackgroundLocation
                                        .setAndroidNotification(
                                      title: 'Background service is running',
                                      message:
                                          'Background location in progress',
                                      icon: '@drawable/ic_notification',
                                    );
                                    await bg.BackgroundLocation
                                        .setAndroidNotification(
                                      title: 'Background service is running',
                                      message:
                                          'Background location in progress',
                                      icon: '@drawable/ic_notification',
                                    );
                                    //await BackgroundLocation.setAndroidConfiguration(1000);
                                    await bg.BackgroundLocation
                                        .startLocationService(
                                            distanceFilter: 200);
                                    await bg.BackgroundLocation
                                        .startLocationService(
                                            distanceFilter: 200);
                                    bg.BackgroundLocation.getLocationUpdates(
                                        (location) {
                                      print("Location updated"+location.toString()); 
                                    });
}
)`

When clicked on the button that calls the service, no error is printed and works fine when clicked for second time.

My flutter doctor is below

`Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, 2.0.6, on Microsoft Windows [Version 10.0.14393], locale en-IN)
[√] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
[√] Chrome - develop for the web
[√] Android Studio (version 4.1.0)
[√] VS Code (version 1.57.1)
[√] Connected device (2 available)`

**Comment by luis901101** [comment]
Same happen to me. This bug begun in version [^0.6.1](https://pub.dev/packages/background_location/versions/0.6.1), before that version the notification starts at the first call.

**Comment by luis901101** [comment]
Digging a bit more I found that the issue is not only the notification not showing at first call but also the background location not starting. 
When app goes to background the notification should startForeground as an ongoing notification _(to ensure intent service long live)_, and background location listener should acquire GPS. When all of this is OK then the app notification shows up and the GPS symbol shows up as well proving the app is in deed receiving background locations.
**In version [0.6.1](https://pub.dev/packages/background_location/versions/0.6.1) this is not working well.**

Some examples for better understanding of the issue:
- App working fine using version **0.6.0**  [View example](https://www.dropbox.com/s/71umtt3d9eodjal/0.6.0.mp4?dl=0). Note when app goes to background the notification and GPS symbol shows up immediately.
- App not working using version **0.6.1**  [View example](https://www.dropbox.com/s/jnzts0isf0occgi/0.6.1-same-code.mp4?dl=0). The code is exactly the same as previous example, note when app goes to background nor the notification nor GPS symbol shows up.
- App not working using version **0.6.1** with notification workaround  [View example](https://www.dropbox.com/s/4ikeybeo2pvqezu/0.6.1-workaround.mp4?dl=0). Here the only change in code is calling the notification twice, note when app goes to background the notification shows up but GPS symbol does not.


**Note: the app in the examples uses [geolocator ^7.2.0+1](https://pub.dev/packages/geolocator/versions/7.2.0+1) for foreground locations**

**Comment by blaqkrow** [comment]
Hey man so if you switch back to 0.6.0 it worked? Im having trouble receiving callbacks when I navigate away from the app and into the background. Heres my code: https://github.com/blaqkrow/location_tracking_flutter

**Comment by luis901101** [comment]
> Hey man so if you switch back to 0.6.0 it worked? Im having trouble receiving callbacks when I navigate away from the app and into the background. Heres my code: https://github.com/blaqkrow/location_tracking_flutter

Yes, using version 0.6.0 works fine, to ensure using that especific version you should set **background_location: 0.6.0** in your pubspec, the tip here is not to use the ^ at the begining of the version.
I see in your code you don’t set the Android notification, I recommend you to set it, as described in the README of this plugin, await for the notification and then start location service.

**Comment by DevTiago** [comment]
The answer from @luis901101 saved my day: just change the version from pubspec.yaml from `^0.6.1` to `0.6.0`

**Comment by yanlok** [comment]
Hi, I would like to ask if this background service not running on first call being fixed in latest version, because I have an issue like this where, during first initial it does not get background updates. Only when I kill the app and open back, it starts getting live background updates
